test/unit/org/apache/cassandra/io/sstable/SSTableWriterTest.java

```diff
@@ -245,4 +249,60 @@ public void testValueTooBigCorruption() throws InterruptedException
         }
     }
 
+    private static void assertValidRepairMetadata(long repairedAt, UUID pendingRepair, boolean isTransient)
+    {
+        Keyspace keyspace = Keyspace.open(KEYSPACE);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF_SMALL_MAX_VALUE);
+        File dir = cfs.getDirectories().getDirectoryForNewSSTables();
+        LifecycleTransaction txn = LifecycleTransaction.offline(OperationType.STREAM);
+
+        try (SSTableWriter writer = getWriter(cfs, dir, txn, repairedAt, pendingRepair, isTransient))
+        {
+            // expected
+        }
+        catch (IllegalArgumentException e)
+        {
+            throw new AssertionError("Unexpected IllegalArgumentException", e);
+        }
+
+        txn.abort();
+        LifecycleTransaction.waitForDeletions();
+    }
+
+    private static void assertInvalidRepairMetadata(long repairedAt, UUID pendingRepair, boolean isTransient)
+    {
+        Keyspace keyspace = Keyspace.open(KEYSPACE);
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF_SMALL_MAX_VALUE);
+        File dir = cfs.getDirectories().getDirectoryForNewSSTables();
+        LifecycleTransaction txn = LifecycleTransaction.offline(OperationType.STREAM);
+
+        try (SSTableWriter writer = getWriter(cfs, dir, txn, repairedAt, pendingRepair, isTransient))
+        {
+            fail("Expected IllegalArgumentException");
+        }
+        catch (IllegalArgumentException e)
+        {
+            // expected
+        }
+
+        txn.abort();
+        LifecycleTransaction.waitForDeletions();
+    }
+
+    /**
+     * It should only be possible to create sstables marked transient that also have a pending repair
+     */
+    @Test
+    public void testRepairMetadataValidation()
+    {
+        assertValidRepairMetadata(UNREPAIRED_SSTABLE, NO_PENDING_REPAIR, false);
+        assertValidRepairMetadata(1, NO_PENDING_REPAIR, false);
+        assertValidRepairMetadata(UNREPAIRED_SSTABLE, UUID.randomUUID(), false);
+        assertValidRepairMetadata(UNREPAIRED_SSTABLE, UUID.randomUUID(), true);
+
+        assertInvalidRepairMetadata(UNREPAIRED_SSTABLE, NO_PENDING_REPAIR, true);
```

Comment:
It means it was part of a transient range at the time the repair began, and should be discarded when the repair session completes successfully. Incremental repairs persist across restarts, although they can't be resumed. 

Since compaction is used to promote/demote sstables, it's possible to finish an incremental repair, but crash before the sstables are able to be promoted. When we restart, we need to be able to figure out the right thing to do with those sstables if we expect the repaired data set to be consistent across nodes.